src/GridInterpolations.jl

```diff
@@ -162,6 +162,9 @@ function interpolate(grid::AbstractGrid, data::DenseArray, x::AbstractVector)
 end
 
 function interpolants(grid::RectangleGrid, x::AbstractVector)
+    if any(isnan.(x))
+        throw(DomainError("Input contains NaN!"))
+    end
```

Comment:
This is a very simple implementation which probably allocates a bit. Another way would be iterate over `x` and bail as soon as you find a NaN. Dunno, if worth the complexity though. How large is `x` usually?

Reply:
`x` has the same size as the grid dimension so not too large. I think that is fine!

Reply:
@rejuvyesh `any(isnan, x)` does not have to allocate (#27). (Most reducers in julia take a function first argument, and that seems to be the most optimized form). I also tried the following (with somewhat confusing results):
- `any(isnan(t) for t in x)`: no allocations
- `any(map(isnan, x))`: allocates, but much faster than `any(isnan.(x))`
None of them allocate anything if x is a StaticVector